### PR TITLE
Better type for directives

### LINKED ISSUE #224: Type errors in `repeat.ts`
Don't know, I'm not really a TypeScripter, but Code squiggles

image

### PR DESCRIPTION
**Pull request by kaste**
Fixes #224 

The patch is not as small as I intended bc I had to propagate the changes to the available directives.

### CODE REVIEW

**Comment by kaste** [comment]
I still think something like 

https://github.com/PolymerLabs/lit-html/blob/5742a9e5e6ae74ee91b0421bca92d90f3e2b0601/src/lit-html.ts#L352-L354

should warn, bc we said directives are functions to `void`. (It should also warn if we return something from a directive.)

File: src/lit-html.ts [hunk 1]
```diff
@@ -356,10 +356,13 @@ export const getValue = (part: Part, value: any) => {
   return value === null ? undefined : value;
 };
 
-export type DirectiveFn<P extends Part = Part> = (part: P) => any;
+export interface DirectiveFn<P=Part> {
+  (part: P): void;
+  __litDirective?: boolean;
```
Comment (justinfagnani) [inline src/lit-html.ts]: You should be able to do: `__litDirective?: true`;

**Comment by kaste** [comment]
Rebased and addressed feedback

**Comment by justinfagnani** [comment]
Thanks @kaste! Sorry this took so long to merge!

**Pull request merged**